### PR TITLE
Fix part of #1460: Exploration file names updated

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR fixes part of #1460.

## Checklist
- [x] Introduce `topics.json` file so that we can dynamically load explorations.
- [x] Renames all exploration files based on `exploration_id` so that the code be made simpler.
- [x] Structure of exploration json files needs to be updated.

## How to test:
Run the app and check if explorations are working or not.
To test if all explorations are still visible correctly working or not, in `HomeFragmentPresenter`, add this line in `handleCreateView`. Note, you will need to inject `StoryProgressTestHelper` in constructor.

```
 logHomeActivityEvent()
 .
 .
 storyProgressTestHelper.markFullTopicProgressForFractions(profileId, false)
 storyProgressTestHelper.markTwoPartialStoryProgressForRatios(profileId, false)
 .
 .
 welcomeViewModel = WelcomeViewModel()
```

## Pending (will be done in future PRs)
- [ ] Skills json file needs to be generalised. Currently we are using `fractions_skills.json` and we need to shift this to id based naming.
- [ ] Questions json files needs to be generalised.
- [ ] Structure of question json files needs to be updated.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt [hunk 1]
```diff
@@ -32,39 +20,23 @@ class ExplorationRetriever @Inject constructor(
...
-      return Exploration.newBuilder()
-        .setId(explorationObject.getString("exploration_id"))
-        .setTitle(explorationObject.getString("title"))
-        .setLanguageCode(explorationObject.getString("language_code"))
-        .setInitStateName(explorationObject.getString("init_state_name"))
-        .setObjective(explorationObject.getString("objective"))
-        .putAllStates(createStatesFromJsonObject(explorationObject.getJSONObject("states")))
-        .build()
-    } catch (e: IOException) {
-      exceptionLogger.logException(e)
```
Comment (anandwana001) [inline domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt:65]: Could we remove `exceptionLogger` from the constructor `private val exceptionLogger: ExceptionLogger`,  as we are no more using it?

**Review by anandwana001 — APPROVED** [review]
LGTM, 
All explorations are working correctly, just on nit change mention in review pass. 
Also, In the fourth exploration, the question `what are the primary colors of light?` is not having a correct answer!! I tried selecting 

```
"<p>Red</p>",
"<p>Blue</p>",
"<p>Green</p>"
```
but no feedback after that. Is there any other answer to this question?
```
"rule_specs": [
                {
                  "inputs": {
                    "x": [
                      "<p>Red</p>",
                      "<p>Blue</p>",
                      "<p>Green</p>"
                    ]
                  },
                  "rule_type": "Equals"
                }
              ],
```
Screenshot_1595231352

One HTML line is not parsing, not related to this PR but mention here to have a reference.
`"html": "<div>Let <oppia-noninteractive-math raw_latex-with-value=\"&quot;x=11&quot;\"></oppia-noninteractive-math>. What is the value of <oppia-noninteractive-math raw_latex-with-value=\"&quot;x^2&quot;\"></oppia-noninteractive-math>?</div><div><br></div>"`
This line is not parsing properly. 
Screenshot_1595229153

File: domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt (hunk 1 above)
Comment (rt4914) [reply domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt:65]: Removed.

Comment (anandwana001) [reply domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt:65]: Thanks

**Comment by rt4914** [comment]
This is working correctly.

Screenshot_1595241157

Also, `<oppia-interactive-math>` is currently not supported in android app.

**Comment by anandwana001** [comment]
> This is working correctly.
> 
> Screenshot_1595241157
> 
> Also, `<oppia-interactive-math>` is currently not supported in android app.

It's in `Fourth Exploration`.

**Comment by rt4914** [comment]
@anandwana001 My last commit updates the exploration such that, the latex tag is removed as we won't be adding support for that and issue mentioned in above screenshot is also solved. Have a look.

**Comment by anandwana001** [comment]
> @anandwana001 My last commit updates the exploration such that, the latex tag is removed as we won't be adding support for that and issue mentioned in the above screenshot is also solved. Have a look.

Working perfectly now, Thank you for the update.

**Pull request merged**